### PR TITLE
IBX-181: Rewrite query and controller documentation

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-181
| Versions      | 3.x

### CODE REVIEW

File: code_samples/front/embed_content/src/Controller/RelationController.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public function __construct(ContentService $contentService, LocationService $locationService)
+    {
+        $this->contentService = $contentService;
+        $this->locationService = $locationService;
+    }
+
+    public function showContentAction(View $view, $locationId)
+    {
+        $accepted_content_types = $view->getParameter('accepted_content_types');
```
Comment (ViniTou) [inline code_samples/front/embed_content/src/Controller/RelationController.php:24]:
```suggestion
        $acceptedContentTypes = $view->getParameter('accepted_content_types');
```

File: code_samples/front/embed_content/src/Controller/RelationController.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        $location = $this->locationService->loadLocation($locationId);
+        $contentInfo = $location->getContentInfo();
+        $versionInfo = $this->contentService->loadVersionInfo($contentInfo);
+        $relations = $this->contentService->loadRelations($versionInfo);
+
+        $items = [];
+
+        foreach ($relations as $relation) {
+            if (in_array($relation->getDestinationContentInfo()->getContentType()->identifier, $accepted_content_types)) {
```
Comment (ViniTou) [inline code_samples/front/embed_content/src/Controller/RelationController.php:34]:
```suggestion
            if (in_array($relation->getDestinationContentInfo()->getContentType()->identifier, $acceptedContentTypes)) {
```

File: docs/guide/content_rendering/queries_and_controllers/controllers.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Configuring a controller enables you to modify and enhance the way in which the built-in content view controller renders content.
+
+You indicate which controller to use in the [content view configuration](../templates/template_configuration.md), under the `controller` key:
+
+``` yaml
+[[= include_file('code_samples/front/embed_content/config/packages/views.yaml', 23, 26) =]][[= include_file('code_samples/front/embed_content/config/packages/views.yaml', 28, 30) =]]
+```
+
+A custom controller must extend `eZ\Bundle\EzPublishCoreBundle\Controller`.
```
Comment (ViniTou) [inline docs/guide/content_rendering/queries_and_controllers/controllers.md:11]: Is this really true?

File: code_samples/front/custom_query_type/config/packages/views.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+ezdesign:
+    design_list:
+        my_design: [ my_theme ]
```
Comment (ViniTou) [inline code_samples/front/custom_query_type/config/packages/views.yaml:3]: is creating and assigning necessary? If not I think it just clouds this example.

**Review by ViniTou — COMMENTED** [review]
And as mentioned, php/temlates files are lacking empty line at the end

File: docs/guide/content_rendering/queries_and_controllers/controllers.md (hunk 3 above)
Comment (DominikaK) [reply docs/guide/content_rendering/queries_and_controllers/controllers.md:11]: Ah, I was hoping you'd know ;)

File: code_samples/front/custom_query_type/config/packages/views.yaml (hunk 4 above)
Comment (DominikaK) [reply code_samples/front/custom_query_type/config/packages/views.yaml:3]: This part of the file will not be displayed in the doc, but it's included in the file so we can easily test the examples

File: docs/guide/content_rendering/queries_and_controllers/controllers.md (hunk 3 above)
Comment (ViniTou) [reply docs/guide/content_rendering/queries_and_controllers/controllers.md:11]: I think it is not needed but let me check then :D

Comment (ViniTou) [reply docs/guide/content_rendering/queries_and_controllers/controllers.md:11]: @DominikaK 
I now can confirm, that this controller does not have to extend anything in order to this to work.

File: code_samples/front/embed_content/src/Controller/RelationController.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\Core\MVC\Symfony\View\View;
+
+class RelationController extends Controller
```
Comment (ViniTou) [inline code_samples/front/embed_content/src/Controller/RelationController.php:10]: __
```suggestion
class RelationController
```

File: code_samples/front/embed_content/src/Controller/RelationController.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
```
Comment (ViniTou) [inline code_samples/front/embed_content/src/Controller/RelationController.php:5]:
```suggestion
```

**Review by ViniTou — APPROVED** [review]
We can also simplify code related to controller.

File: docs/guide/content_rendering/queries_and_controllers/controllers.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+Configuring a controller enables you to modify and enhance the way in which the built-in content view controller renders content.
+
+You indicate which controller to use in the [content view configuration](../templates/template_configuration.md), under the `controller` key:
+
+``` yaml
+[[= include_file('code_samples/front/embed_content/config/packages/views.yaml', 23, 26) =]][[= include_file('code_samples/front/embed_content/config/packages/views.yaml', 28, 30) =]]
+```
+
+``` php
+[[= include_file('code_samples/front/embed_content/src/Controller/RelationController.php', 2, 10) =]]
```
Comment (DominikaK) [inline docs/guide/content_rendering/queries_and_controllers/controllers.md:12]:
```suggestion
[[= include_file('code_samples/front/embed_content/src/Controller/RelationController.php', 2, 9) =]]
```

File: docs/guide/content_rendering/queries_and_controllers/content_queries.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+### Content view configuration
+
+To render a Content query Field, in the content view configuration use the `content_query_field` view type:
+
+``` yaml
+[[= include_file('code_samples/front/list_content/config/packages/views.yaml', 8, 9) =]][[= include_file('code_samples/front/list_content/config/packages/views.yaml', 22, 28) =]]
+```
+
+The identifier of the Content query Field must be matched
```
Comment (dabrt) [inline docs/guide/content_rendering/queries_and_controllers/content_queries.md:148]:
```suggestion
The identifier of the Content query Field must be matched by
```

File: docs/guide/content_rendering/queries_and_controllers/controllers.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Controllers
+
+Configuring a controller enables you to modify and enhance the way in which the built-in content view controller renders content.
```
Comment (dabrt) [inline docs/guide/content_rendering/queries_and_controllers/controllers.md:3]:
```suggestion
By configuring a controller you can modify and enhance the way in which the built-in content view controller renders content.
```

File: docs/guide/content_rendering/queries_and_controllers/custom_query_type.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+``` php
+[[= include_file('code_samples/front/custom_query_type/src/QueryType/LatestContentQueryType.php', 10, 14) =]]
+```
+
+!!! caution
+
+    Query type name must be unique.
+
+The `getQuery()` method constructs the query, based on Search Criteria and Sort Clauses.
```
Comment (dabrt) [inline docs/guide/content_rendering/queries_and_controllers/custom_query_type.md:30]:
```suggestion
The `getQuery()` method constructs the query based on Search Criteria and Sort Clauses.
```

File: docs/guide/content_rendering/queries_and_controllers/custom_query_type.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+See [Content Search](../../../api/public_php_api_search.md) for more information about queries
```
Comment (dabrt) [inline docs/guide/content_rendering/queries_and_controllers/custom_query_type.md:31]:
```suggestion
See [Content search](../../../api/public_php_api_search.md) for more information about queries
```

File: docs/guide/content_rendering/queries_and_controllers/custom_query_type.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+In the content view configuration, indicate that the content view should use the custom Query type:
+
+``` hl_lines="10"
+[[= include_file('code_samples/front/custom_query_type/config/packages/views.yaml', 8, 21) =]]
+```
+
+## Options resolver-based Query type
+
+Additionally, your custom Query type can use extend the `OptionsResolverBasedQueryType` abstract class.
```
Comment (dabrt) [inline docs/guide/content_rendering/queries_and_controllers/custom_query_type.md:45]:
```suggestion
Additionally, your custom Query type can extend the `OptionsResolverBasedQueryType` abstract class.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/guide/content_rendering/embed_and_list_content/list_content.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+This template uses the [`ez_render()` Twig function](../twig_function_reference/content_twig_functions.md#ez_render)
+to render every child of the folder with the default template for the `line` view.
+
+## List children in Content query Field
+
+A [Content query Field](../../../api/field_types_reference/contentqueryfield.md) is a Field that defines a query.
+The following example shows how to use a Content query Field to render a Blog with its Blog Post children.
+
+Create a Blog Content Type that contains a Content query Field with the identifier `query`.
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/guide/content_rendering/embed_and_list_content/list_content.md:44]: That's a good point, and I think I'll try it out, but as it should also apply to the procedures from the previous PR, I'll handle it in the joint "cleanup" PR before merging the feature branch to master.

**Review by dabrt — APPROVED** [review]

**Pull request merged**